### PR TITLE
fix(Purse): #MLYSTOR-711 fix update purse

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes
fix update purse and code upgrade 
## Checklist tests
-update purse
-import purse 

## Issue ticket number and link
https://jira.support-ent.fr/browse/MLYSTOR-711
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/lystore/controllers/PurseController.java [hunk 1]
```diff
@@ -255,7 +253,9 @@ private void returnErrorMessage(HttpServerRequest request, Throwable cause, Stri
      * @param cause   Cause error
      */
     private static void renderErrorMessage(HttpServerRequest request, Throwable cause) {
-        renderError(request, new JsonObject().put("message", cause.getMessage()));
+
+        badRequest(request, cause.getMessage());
+//        renderError(request, new JsonObject().put("message", cause.getMessage()));
```
Comment (FlorentMr) [inline src/main/java/fr/openent/lystore/controllers/PurseController.java:258]: com en trop

File: src/main/resources/public/ts/controllers/administration/purse.ts [hunk 2]
```diff
@@ -1,95 +1,170 @@
...
+    exportPurses(id: number): void;
+
+    checkPurses(id_Campaign: number): void;
+}
+
+interface IMainScope extends IScope {
+    vm: IViewModel;
+}
+
+// class LystoreAxiosError  extends AxiosError{
```
Comment (FlorentMr) [inline src/main/resources/public/ts/controllers/administration/purse.ts:59]: com inutile?

File: src/main/resources/public/ts/controllers/administration/purse.ts [hunk 3]
```diff
@@ -1,95 +1,170 @@
...
-            window.location = `/lystore/campaign/${id}/purses/export`;
-        };
-        $scope.checkPurses = async ( id_Campaign : number ) =>{
-            $scope.isChecked = true;
-            await $scope.campaign.purses.check(id_Campaign);
-            Utils.safeApply($scope)
-        }
+    };
+
+    exportPurses = (id: number) => {
```
Comment (FlorentMr) [inline src/main/resources/public/ts/controllers/administration/purse.ts:160]: retour void

File: src/main/resources/public/ts/controllers/administration/purse.ts [hunk 4]
```diff
@@ -1,95 +1,170 @@
...
+        window.location = `/lystore/campaign/${id}/purses/export`;
+    };
+    checkPurses = async (id_Campaign: number) => {
```
Comment (FlorentMr) [inline src/main/resources/public/ts/controllers/administration/purse.ts:163]: retour void

File: src/main/resources/public/ts/model/Purse.ts [hunk 5]
```diff
@@ -1,82 +1,90 @@
...
+import {Structure} from "./Structure";
+
+
+export interface IPurseStructureResponse {
+    id: number,
+    amount: number,
+    id_campaign: number,
+    initial_amount: number,
+    total_order: number,
+    uai: string,//à changer en Istruct surement
```
Comment (FlorentMr) [inline src/main/resources/public/ts/model/Purse.ts:13]: encore d'actualité?

File: src/main/resources/public/ts/model/Purse.ts [hunk 6]
```diff
@@ -1,82 +1,90 @@
...
-                        }
-                    }
-                })
-            })
-        }
+    build(pursesData: IPurseStructureResponse[]): Purses {
+        this.all = pursesData.map(purseData => {
+            return new Purse().build(purseData);
+        })
+        return this
```
Comment (FlorentMr) [inline src/main/resources/public/ts/model/Purse.ts:87]: manque ; (oui je suis relou dsl)

File: src/main/resources/public/ts/services/app/admin/purse.service.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    },
+
+    sync(idCampaign: number): Promise<Purses> {
+        return http.get(`/lystore/campaign/${idCampaign}/purses/list`).then((res: AxiosResponse) => {
+            let PurseStructureResponses: IPurseStructureResponse[] = res.data;
+            return new Purses(idCampaign).build(PurseStructureResponses);
+        }).catch(e => {
+            console.error(e);
+            notify.error("lystore.purse.get.err")
+            return new Purses(idCampaign)
```
Comment (FlorentMr) [inline src/main/resources/public/ts/services/app/admin/purse.service.ts:29]: là aussi des ptits ;

**Review by qpic-cgi — APPROVED** [review]

**Review by FlorentMr — APPROVED** [review]

**Review by lapeyronnie — APPROVED** [review]

**Pull request merged**